tests/baselines/reference/unionsOfTupleTypes1.errors.txt

```diff
@@ -41,18 +41,18 @@ tests/cases/conformance/types/tuple/unionsOfTupleTypes1.ts(41,18): error TS2339:
     function f1(t1: T1, t2: T2, t3: T3, t4: T4, x: number) {
         let [d10, d11, d12] = t1;  // string, number
                        ~~~
-!!! error TS2493: Tuple type '[string, number]' with length '2' cannot be assigned to tuple with length '3'.

```

Comment:
This error message change is a regression, I believe. An explicit tuple length error here is quite desirable.

Reply:
I think the new error is just as good, if not better, and certainly more consistent. I always found this particular message strange as there isn't actually a tuple being assigned to. If we think it is important, we could instead improve the message we give when accessing an out-of-range tuple element in `getIndexedAccessType`.

Reply:
We _explicitly_ added the tuple length error message to override the message it regressed to. Changing `getIndexedAccessType` to use it seems kinda appropriate.

Reply:
I think it would be nice to give a more specific error message if it can still be rewired into `getIndexedAccessType`:

```
Tuple type '{0}' of length '{1}' has no element at index '{2}'.

Seems pretty reasonable.

Reply:
I think, like Daniel, that will be better message to avoid issues like #29058.